kotoed-js/src/main/ts/code/actions.ts

```diff
@@ -322,28 +357,38 @@ export function loadFileToEditor(payload: FilePathPayload & SubmissionPayload) {
             file: filename
         })(dispatch, getState);
 
-        fetchFile(payload.submissionId, filename).then(result => {
-            dispatch(fileLoad.done({
-                params: {
-                    submissionId: payload.submissionId,
-                    filename
-                },
-                result: {
-                    value: result,
-                    displayedComments: getState().commentsState.comments.get(filename, Map<number, LineComments>()),
-                }
-            }));
-        });
+        const result = await fetchFile(payload.submissionId, filename);
+        dispatch(fileLoad.done({
+            params: {
+                submissionId: payload.submissionId,
+                filename
+            },
+            result: {
+                value: result,
+                displayedComments: state.commentsState.comments.get(filename, Map<number, LineComments>()),
+            }
+        }));
 
-        fetchDiff(payload.submissionId).then(result => {
-            dispatch(fileDiff.done({
-                params: {
-                    submissionId: payload.submissionId,
-                    filename
-                },
-                result: result.find((diff) => diff.toFile == filename)
```

Comment:
Diff now is fetched only once, since diff endpoint returns a diff for the whole submission.